src/components/table/utils.js

```diff
@@ -64,3 +64,29 @@ export const getValidRows = ({ selectedFlatRows, isGrouped, itemIsDisabled }) =>
     acc.push(row.original)
     return acc
   }, [])
+
+export const getGroupedCells = cells => cells.reduce((accumulator, cell) => {
```

Comment:
Here as well you can create separate function for each of these two cases. Right now it is not very clear what is getting returned in each case.